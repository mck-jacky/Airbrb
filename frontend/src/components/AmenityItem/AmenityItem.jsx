import React from 'react';
import PropTypes from 'prop-types';

import AcUnitOutlinedIcon from '@mui/icons-material/AcUnitOutlined';
import BathtubOutlinedIcon from '@mui/icons-material/BathtubOutlined';
import ChairOutlinedIcon from '@mui/icons-material/ChairOutlined';
import CheckroomOutlinedIcon from '@mui/icons-material/CheckroomOutlined';
import ConnectedTvOutlinedIcon from '@mui/icons-material/ConnectedTvOutlined';
import CountertopsOutlinedIcon from '@mui/icons-material/CountertopsOutlined';
import DryOutlinedIcon from '@mui/icons-material/DryOutlined';
import FireplaceOutlinedIcon from '@mui/icons-material/FireplaceOutlined';
import HttpsOutlinedIcon from '@mui/icons-material/HttpsOutlined';
import HvacOutlinedIcon from '@mui/icons-material/HvacOutlined';
import IronOutlinedIcon from '@mui/icons-material/IronOutlined';
import LocalLaundryServiceOutlinedIcon from '@mui/icons-material/LocalLaundryServiceOutlined';
import RestaurantOutlinedIcon from '@mui/icons-material/RestaurantOutlined';
import StadiumOutlinedIcon from '@mui/icons-material/StadiumOutlined';
import TvOutlinedIcon from '@mui/icons-material/TvOutlined';
import WavesOutlinedIcon from '@mui/icons-material/WavesOutlined';
import { Typography } from '@mui/material';

const AmenityItem = ({ amenity }) => {
  const amenityStyles = {
    fontSize: 20,
    position: 'relative',
    top: 5,
    marginRight: '8px'
  };

  let iconComponent = <></>
  if (amenity === 'essentials') {
    iconComponent = <RestaurantOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'air-conditioning') {
    iconComponent = <AcUnitOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'hair-dryer') {
    iconComponent = <DryOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'iron') {
    iconComponent = <IronOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'dryer') {
    iconComponent = <StadiumOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'TV') {
    iconComponent = <ConnectedTvOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'indoor-fireplace') {
    iconComponent = <FireplaceOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'private-entrance') {
    iconComponent = <HttpsOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'kitchen') {
    iconComponent = <CountertopsOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'heating') {
    iconComponent = <HvacOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'hangers') {
    iconComponent = <CheckroomOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'washer') {
    iconComponent = <LocalLaundryServiceOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'hot-water') {
    iconComponent = <WavesOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'cable-TV') {
    iconComponent = <TvOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'private-bathroom') {
    iconComponent = <BathtubOutlinedIcon sx={amenityStyles} />
  } else if (amenity === 'private-living-room') {
    iconComponent = <ChairOutlinedIcon sx={amenityStyles} />
  }

  return (
    <>
    <Typography data-testid='amenity-item'>
      {iconComponent}
      <span data-testid='amenity-item-title'>{amenity}</span>
    </Typography>
    </>
  )
}

AmenityItem.propTypes = {
  amenity: PropTypes.string
}

export default AmenityItem;
