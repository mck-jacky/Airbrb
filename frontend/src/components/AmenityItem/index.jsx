import AmenityItem from './AmenityItem';

export default AmenityItem
