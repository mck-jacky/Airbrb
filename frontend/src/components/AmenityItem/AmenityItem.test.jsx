import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow'
import AmenityItem from './AmenityItem'

afterEach(() => {
  cleanup();
})

const amenityMap = {
  essentials: 'RestaurantOutlinedIcon',
  'air-conditioning': 'AcUnitOutlinedIcon',
  'hair-dryer': 'DryOutlinedIcon',
  iron: 'IronOutlinedIcon',
  dryer: 'StadiumOutlinedIcon',
  TV: 'ConnectedTvOutlinedIcon',
  'indoor-fireplace': 'FireplaceOutlinedIcon',
  'private-entrance': 'HttpsOutlinedIcon',
  kitchen: 'CountertopsOutlinedIcon',
  heating: 'HvacOutlinedIcon',
  hangers: 'CheckroomOutlinedIcon',
  washer: 'LocalLaundryServiceOutlinedIcon',
  'hot-water': 'WavesOutlinedIcon',
  'cable-TV': 'TvOutlinedIcon',
  'private-bathroom': 'BathtubOutlinedIcon',
  'private-living-room': 'ChairOutlinedIcon'
}

describe('AmenityItem', () => {
  it('is in the document', () => {
    render(<AmenityItem amenity='essentials' />)
    const component = screen.getByTestId('amenity-item')
    expect(component).toBeInTheDocument();
  })

  it('has div', () => {
    render(<AmenityItem amenity='essentials' />)
    const component = screen.getByTestId('amenity-item')
    expect(component).toContainHTML('<p ')
  })

  for (const key of Object.keys(amenityMap)) {
    it('has icon', () => {
      render(<AmenityItem amenity={key} />)
      const component = screen.getByTestId(amenityMap[key])
      expect(component).toBeInTheDocument();
    })
  }

  for (const key of Object.keys(amenityMap)) {
    it('has title', () => {
      render(<AmenityItem amenity={key} />)
      const component = screen.getByTestId('amenity-item-title')
      expect(component).toBeInTheDocument();
      expect(component).toHaveTextContent(key)
    })
  }

  it('has to match the snapshot', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<AmenityItem amenity='essentials' />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
