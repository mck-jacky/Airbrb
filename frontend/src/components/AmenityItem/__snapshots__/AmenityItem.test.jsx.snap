// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmenityItem has to match the snapshot 1`] = `
<React.Fragment>
  <ForwardRef(Typography)
    data-testid="amenity-item"
  >
    <Memo
      sx={
        Object {
          "fontSize": 20,
          "marginRight": "8px",
          "position": "relative",
          "top": 5,
        }
      }
    />
    <span
      data-testid="amenity-item-title"
    >
      essentials
    </span>
  </ForwardRef(Typography)>
</React.Fragment>
`;
